weave/wandb_interface/wandb_stream_table.py

```diff
@@ -1,94 +1,18 @@
 import atexit
-import contextlib
-import datetime
 import json
-import logging
-import os
 import queue
-import random
-import shutil
 import threading
 import typing
 import uuid
 
-from wandb.sdk.lib.paths import LogicalPath
-from wandb.sdk.lib.printer import get_printer
-from wandb.sdk.lib.ipython import _get_python_type
+from wandb import errors
+from wandb.sdk.artifacts.artifact import Artifact
 
+from .lib.ipython import _get_python_type
+from .lib.printer import get_printer
+from .wandb_lite_run import InMemoryLazyLiteRun, wandb_public_api
 
-from .wandb_lite_run import InMemoryLazyLiteRun
-
-from .. import runfiles_wandb
-from .. import storage
-from .. import weave_types
-from .. import artifact_base
-from .. import environment
-from .. import file_util
-from .. import graph
-from .. import errors
-from ..core_types.stream_table_type import StreamTableType
-from ..ops_domain import stream_table_ops
-from ..ops_primitives import weave_api
-
-if typing.TYPE_CHECKING:
-    from wandb.sdk.internal.file_pusher import FilePusher
-
-
-# Shawn recommended we only encode leafs, but in my testing, nested structures
-# are not handled as well in in gorilla and we can do better using just weave.
-# Uncomment the below to use gorilla for nested structures.
-TRUST_GORILLA_FOR_NESTED_STRUCTURES = True
-
-# Weave types are parametrized, but gorilla expects just simple strings. We could
-# send the top-level string over the wire, but this fails to encode type specifics
-# and therefore loses information. With this flag, we instead stringify the JSON type
-# and send that over the wire. This is a bit of a hack, but it works.
-ENCODE_ENTIRE_TYPE = True
-TYPE_ENCODE_PREFIX = "_wt_::"
-
-ROW_TYPE = typing.Union[dict, list[dict]]
-
-
-class WandbLiveRunFiles(runfiles_wandb.WandbRunFiles):
-    _file_pusher: typing.Optional["FilePusher"] = None
-    _temp_dir: typing.Optional[str] = None
-
-    def set_file_pusher(self, pusher: "FilePusher") -> None:
-        # It is the responsibility of the caller to ensure this file pusher is
-        # correctly associated with the corresponding run.
-        self._file_pusher = pusher
-
-    def temp_dir(self) -> str:
-        if self._temp_dir is None:
-            rand_part = "".join(random.choice("0123456789ABCDEF") for _ in range(16))
-            self._temp_dir = os.path.join(
-                runfiles_wandb.wandb_run_dir(), "upload_cache", f"tmp_{rand_part}"
-            )
-            os.makedirs(self._temp_dir, exist_ok=True)
-        return self._temp_dir
-
-    def cleanup(self) -> None:
-        if self._temp_dir is not None:
-            shutil.rmtree(self._temp_dir)
-            self._temp_dir = None
-
-    def __del__(self) -> None:
-        self.cleanup()
```

Comment:
I didn't test the file run file uploading stuff.  I assume this was only there for weave artifacts.  I run a file_pusher in my live_run interface...  I should have a test that logs an image... 🙃